src/hotspot/share/runtime/nonJavaThread.hpp

```diff
@@ -103,6 +103,15 @@ class WorkerThread: public NamedThread {
  private:
   uint _id;
  public:
+  static WorkerThread* current() {
+    return WorkerThread::cast(Thread::current());
+  }
+
+  static WorkerThread* cast(Thread* t) {
+    assert(t->is_Worker_thread(), "incorrect cast to const WorkerThread");
```

Comment:
It should probably not say *const* here.

Reply:
`as_Worker_thread()` used const so I wonder whether the new cast function should too?

Reply:
You mean `static const WorkerThread* cast(const Thread* t) { ... }`? As the implementation of this method is nothing but a `static_cast` (and it will probably stay that way in the future), having `const Thread*` won't really catch anything interesting, IMO.